crates/models/llama/src/lib.rs

```diff
@@ -64,39 +53,49 @@ impl KnownModel for Llama {
-            Backend::Gpu
-        } else {
-            Backend::Cpu
-        };
+        let backend = params.backend(0);
 
-        let norm = tl.offload("norm.weight", backend)?;
+        let norm = tl.load("norm.weight")?.transfer_to(backend)?;
 
-        let output = tl.offload("output.weight", backend)?;
+        let output = tl.load("output.weight")?.transfer_to(backend)?;
 
         let mut layers = Vec::new();
 
         for i in 0..hyperparameters.n_layer {
-            let backend = if params.should_offload(i) {
-                Backend::Gpu
-            } else {
-                Backend::Cpu
-            };
+            let backend = params.backend(i);
+
             let layer = Layer {
                 attention_norm: tl
-                    .offload(&format!("layers.{i}.attention_norm.weight"), backend)?,
-                wq: tl.offload(&format!("layers.{i}.attention.wq.weight"), backend)?,
-                wk: tl.offload(&format!("layers.{i}.attention.wk.weight"), backend)?,
-                wv: tl.offload(&format!("layers.{i}.attention.wv.weight"), backend)?,
-                wo: tl.offload(&format!("layers.{i}.attention.wo.weight"), backend)?,
-                ffn_norm: tl.offload(&format!("layers.{i}.ffn_norm.weight"), backend)?,
-                w1: tl.offload(&format!("layers.{i}.feed_forward.w1.weight"), backend)?,
-                w2: tl.offload(&format!("layers.{i}.feed_forward.w2.weight"), backend)?,
-                w3: tl.offload(&format!("layers.{i}.feed_forward.w3.weight"), backend)?,
+                    .load(&format!("layers.{i}.attention_norm.weight"))?
+                    .transfer_to(backend)?,
+                wq: tl
+                    .load(&format!("layers.{i}.attention.wq.weight"))?
+                    .transfer_to(backend)?,
+                wk: tl
+                    .load(&format!("layers.{i}.attention.wk.weight"))?
+                    .transfer_to(backend)?,
+                wv: tl
+                    .load(&format!("layers.{i}.attention.wv.weight"))?
+                    .transfer_to(backend)?,
+                wo: tl
+                    .load(&format!("layers.{i}.attention.wo.weight"))?
+                    .transfer_to(backend)?,
+                ffn_norm: tl
+                    .load(&format!("layers.{i}.ffn_norm.weight"))?
+                    .transfer_to(backend)?,
+                w1: tl
+                    .load(&format!("layers.{i}.feed_forward.w1.weight"))?
+                    .transfer_to(backend)?,
+                w2: tl
+                    .load(&format!("layers.{i}.feed_forward.w2.weight"))?
+                    .transfer_to(backend)?,
+                w3: tl
+                    .load(&format!("layers.{i}.feed_forward.w3.weight"))?
+                    .transfer_to(backend)?,
             };
             layers.push(layer);
         }
-        let (context, loaded_tensors) = tl.finish();
+        let (context, _) = tl.finish();
```

Comment:
It doesn't look like we actually use the loaded tensors in any of the models; maybe we should just remove the second return value, so that it can be `let context = tl.finish();`

Reply:
I guess this also is another cleanup task? I actually don't know why `finish` returns the tensors 😬

Reply:
I'm sure there was a good reason at some point, but we don't need it any more. Up to you if you want to clean it up or not; I can do that otherwise.